gpii/node_modules/flowManager/src/SolutionsRegistryDataSource.js

```diff
@@ -35,45 +43,209 @@ fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource", {
+// from the local file system are used.
+fluid.defaults("gpii.flowManager.solutionsRegistry.dataSource.local", {
+    gradeNames: ["gpii.flowManager.solutionsRegistry.dataSource"],
+    components: {
+        revisionRequester: {
+            type: "gpii.flowmanager.revisionRequester"
+        },
+        repositorySolutionsLoader: {
+            type: "gpii.flowManager.repositorySolutionsLoader"
+        }
+// The platformReporter is distributed down from the local flowManager
+//      platformReporter: {
+//          type: "gpii.platformReporter.native"
+//      }
+    },
+    readOnlyGrade: "gpii.flowManager.solutionsRegistry.dataSource.local",
+    members: {
+        repositorySolutionsRegistry: null
+    },
+    invokers: {
+        getImpl: {
+            funcName: "gpii.flowManager.solutionsRegistry.dataSource.local.handle",
+            args: ["{that}", "{arguments}.1"]
+                             // options
+        }
+    },
+    listeners: {
+        "loadSolutions.loadFromLocalDisk": {
+            listener: "{that}.loadFromLocalDisk",
+            priority: "first"
+        },
+        "loadSolutions.getRevision": {
+            listener: "{revisionRequester}.getRevision",
+            priority: "after:loadFromLocalDisk"
+        },
+        "loadSolutions.loadFromRepository": {
+            listener: "gpii.flowManager.solutionsRegistry.dataSource.local.loadFromRepository",
+            args: ["{that}", "{arguments}.0", "{that}.platformReporter"],
+                             // revision
+            priority: "after:getRevision"
+        },
+        "loadSolutions.solutionsLoaded": {
+            listener: "{that}.events.solutionsRegistryReady",
+            priority: "after:loadFromRepository"
+        }
+    }
+});
+
+/**
+ * Retrieve the solutions registry JSON file from the source code repository.
+ * If successful, sets the repositorySolutionsRegistry member to the fetched
+ * solutions registry.
+ * @param {Component} that - An instance of gpii.flowManager.solutionsRegistry.dataSource
+ * @param {Object} that.repositorySolutionsRegistry - Set to point to the
+ *                 solutions registry retrieved from the respository, or null if
+ *                 none were retrieved.
+ * @param {Object} revision - the SHA256 revision of the repository in the form:
+ *                            { sha256: ... }.  If the sha256 is missing, the
+ *                            result of the load is null.
+ * @param {Component} platformReporter - used to get the name of the platform,
+ *                                       e.g., "win32".
+ * @return {Promise} A promise whose resolved value is eiher the solutions
+ *      registry from the source code repository, or null, if there was an error.
+ */
+gpii.flowManager.solutionsRegistry.dataSource.local.loadFromRepository = function (that, revision, platformReporter) {
+    var gpiiRevision = revision.sha256;
```

Comment:
At the start of the process, it might be worthwhile to check if a proper `revision` is received. If `revision` is not received due to `cloudURL === null`, return and skip the rest.